src/SpendPermissionManager.sol

```diff
@@ -159,26 +207,26 @@ contract SpendPermissionManager is EIP712 {
         emit SpendPermissionRevoked(hash, spendPermission.account, spendPermission);
     }
 
-    /// @notice Approve a spend permission with signature and spend tokens in one call.
+    /// @notice Invalidate a specific batch of spend permissions from being approved.
+    /// @dev This permanently prevents a signature over this specific batch hash from being applied in the future.
+    /// This does not affect the unique spend permissions themselves if they have already been approved
+    /// nor does it prevent the ability to approve them in the future in a different batch or individually.
     ///
-    /// @dev After initially approving a spend permission, it is more gas efficient to call `spend` for repeated use.
-    ///
-    /// @param spendPermission Details of the spend permission.
-    /// @param signature Signed approval from the user.
-    /// @param value Amount of token attempting to spend (wei).
-    function spendWithSignature(SpendPermission memory spendPermission, bytes memory signature, uint160 value)
+    /// @param spendPermissionBatch Details of the batch of spend permissions.
+    function invalidateBatch(SpendPermissionBatch calldata spendPermissionBatch)
```

Comment:
something makes me nervous about this function when we think about what it would be like to build an interface with this.... the scary thing is that even if a user quickly applies this to invalidate the batch they just signed, it's still possible that the signature for the batch had been applied and thus the permissions have been granted.  the call to `invalidateBatch` will succeed (since it's just setting a boolean in a mapping) and the user may erroneously believe that they have successfully invalidated the batch and/or revoked the permissions when in fact the spender still has full access.  

I wonder if it would make sense to check all of the spendPermissions included in the batch and perhaps check if they are not yet approved, and revert if it finds that any of the spend permissions in the batch (which would presumably be either all or none of them) have already been approved with something like `BatchAlreadyApproved` or something.  just so it's at least noisy.

it's cheaper than the revoke approach since this doesn't require storage writes, but it also doesn't actually perform the invalidation or the revoke -- it just lets the user know a bit more about the state of their attempt to invalidate.  and at that point makes me wonder if the revoke approach is actually better because you're not getting any false positives or otherwise confusing UIs